checkapi/internal/diff_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiff(t *testing.T) {
+	a, err := Read(filepath.Join("testpkg", "receiver", "validreceiver"), nil, nil)
+	require.NoError(t, err)
+	b := a
+	bPreConst := b
+	bPreConst.Values = append([]string{"foobar"}, bPreConst.Values...)
+	bAddConst := b
+	bAddConst.Values = append(bAddConst.Values, "foobar")
+	bAddFn := b
+	bAddFn.Functions = append(bAddFn.Functions, Function{
+		Name:        "MyFn",
+		Receiver:    "foo",
+		ReturnTypes: []string{"string", "error"},
+		Params:      []string{"string", "bool"},
+		TypeParams:  []string{"~string"},
+	})
+	bBeforeFn := b
+	bBeforeFn.Functions = append([]Function{{
+		Name:        "Aaaa",
+		Receiver:    "foo",
+		ReturnTypes: []string{"string", "error"},
+		Params:      []string{"string", "bool"},
+	}}, bBeforeFn.Functions...)
+	bAddStruct := b
+	bAddStruct.Structs = append(bAddStruct.Structs, Apistruct{Name: "MyStruct", Fields: []string{"foo", "bar"}})
```

Comment:
It looks like currently you are unable to detect changes related to field types and type parameters, right?

Reply:
you can but it's a bit blunt. I am adding a test for the addition of a field.

If you add a `foobar` field, the result is:

```
				Left: API{
					Structs: []Apistruct{{Name: "MyStruct", Fields: []string{"foo", "bar"}}},
				},
				Equal: API{},
				Right: API{
					Structs: []Apistruct{{Name: "MyStruct", Fields: []string{"foo", "bar", "foobar"}}},
				},
```